### ISSUE TITLE
Service status page

### ISSUE THREAD

**Issue by meh9** [TerriaJS/terriajs#1783]
We have discussed having some form of Service Status page in the past, that shows which services are currently "experiencing problems". We have also had requests from users for something like this, most recently today from somebody in a government agency.

We should consider showing the status in the catalogue UI, or make things unselectable/greyed out or similar.

Obviously the capability of detecting when something is or isn't working differs between different types of services; WMS can have a working `GetCapabilities` but some or all tiles for a layer can fail, some have a failing legend, etc.

Perhaps we should have a `/status-service` where client browsers report failures, and if e.g. 3 failures for a layer is received from clients on different subnets then consider the layer failed, or maybe test failing URL's from the server, or some other scheme like that.

This might then tie into having a Catalogue Service so that when clients request the catalogue they also receive the current status of the layers.

Maybe also register a contact address to send an automated email to?

**Comment by stevage** [comment]
> Perhaps we should have a /status-service where client browsers report failures, and if e.g. 3 failures for a layer is received from clients on different subnets then consider the layer failed, or maybe test failing URL's from the server, or some other scheme like that.

Pretty interesting idea. Where would we store that data then? How would it be queried? How long do statuses last? 

Could be easy to make something fairly basic: 
- report any error (how do we know if we're getting reports from the same person? IP?)
- get a list of URLs with service statuses (eg, error, warning, ok)
- client looks up a URL when instantiating items, to warn of the bad ones.

Basically, we'd be distributing/crowdsourcing a large part of the service QA. Pretty interesting idea. Downside is obviously it's always reactive.

**Comment by meh9** [comment]
Yeah, the technical issues are reasonably significant since we would be storing state in a cluster.

My initial idea was to check by IP but check that they are from different subnets, to increase the likelihood that it's not just a local failure from one particular network.
